platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/util/AddressBookNetworkUtilsTests.java

```diff
@@ -76,4 +83,53 @@ void ErrorOnInvalidLocalAddress() {
         final Address badLocalAddress = address.copySetHostnameInternal("500.8.8");
         assertThrows(IllegalStateException.class, () -> AddressBookNetworkUtils.isLocal(badLocalAddress));
     }
+
+    @Test
+    void testCreateRosterFromNonEmptyAddressBook() {
+        final PlatformBuilder platformBuilder = PlatformBuilder.create(
+                "name",
+                "swirldName",
+                new BasicSoftwareVersion(1),
+                () -> null,
+                (inputStream, path) -> null,
+                new NodeId(0));
+
+        final Address address1 = new Address(new NodeId(1), "", "", 10, null, 77, null, 88, null, null, "");
+        final Address address2 = new Address(new NodeId(2), "", "", 10, null, 77, null, 88, null, null, "");
+        final AddressBook addressBook = new AddressBook();
+        addressBook.add(address1);
+        addressBook.add(address2);
+        platformBuilder.withBootstrapAddressBook(addressBook);
+        final Roster roster = AddressBookUtils.createRoster(addressBook);
+
+        assertNotNull(roster);
+        assertEquals(2, roster.rosters().size());
+        assertEquals(1L, roster.rosters().getFirst().nodeId());
+        assertEquals(2L, roster.rosters().getLast().nodeId());
+    }
+
+    @Test
+    void testCreateRosterFromNullAddressBook() {
+        assertThrows(
+                NullPointerException.class,
+                () -> AddressBookUtils.createRoster(null),
+                "Illegal attempt to create a Roster from a null AddressBook");
+    }
+
+    @Test
+    void testCreateRosterFromEmptyAddressBook() {
+        final PlatformBuilder platformBuilder = PlatformBuilder.create(
+                "name",
+                "swirldName",
+                new BasicSoftwareVersion(1),
+                () -> null,
+                (inputStream, path) -> null,
+                new NodeId(0));
+        final AddressBook addressBook = new AddressBook();
+        platformBuilder.withBootstrapAddressBook(addressBook);
+        final Roster roster = AddressBookUtils.createRoster(addressBook);
```

Comment:
Should this be throwing exceptions?   Does the `createRoster` also need to validate that the roster is valid?

Reply:
I think there's value in separating the creator and validator as that makes it more testable. The core code will create, then validate.